src/cxx/basic_api.cc

```diff
@@ -468,6 +468,79 @@ JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphras
     }
 }
 
+JNIEXPORT jobject JNICALL Java_foundation_pEp_jniadapter_Engine__1unlock_1keys_1with_1passphrase(JNIEnv *env,
+         jobject obj,
+         jobject accountswithpassphrases)
+{
+    std::mutex *mutex_local = nullptr;
+    {
+        std::lock_guard<std::mutex> l(global_mutex);
+        pEpLog("called with lock_guard");
+        mutex_local = get_engine_java_object_mutex(env, obj);
+    }
+    std::lock_guard<std::mutex> l(*mutex_local);
+
+    const stringpair_list_t* _accountswithpassphrases = to_stringpairlist(env, accountswithpassphrases);
+    stringlist_t *_errorAccounts;
```

Comment:
Initialize to nullptr, don't rely on core.

Reply:
Done.